numpy/core/arrayprint.py

```diff
@@ -627,8 +627,11 @@ def __call__(self, x, strip_zeros=True):
 
 def _digits(x, precision, format):
     s = format % x
-    z = s.rstrip('0')
-    return precision - len(s) + len(z)
+    if precision > 0:
+        z = s.rstrip('0')
+        return precision - len(s) + len(z)
+    else:
+        return len(s)
```

Comment:
Shouldn't it just return zero? In the non-zero case there is a decimal point that stops the rstrip so the precision returned is the actual number of digits past the decimal point minus trailing zeros. In the zero case that would always be zero if the decimal point was still there.